src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -356,7 +359,10 @@ protected function instantiateObject(array &$data, $class, array &$context, \Ref
                 } elseif ($constructorParameter->isDefaultValueAvailable()) {
                     $params[] = $constructorParameter->getDefaultValue();
                 } else {
-                    throw new RuntimeException(
+                    if (isset($context[static::EMPTY_DATA][$class])) {
```

Comment:
Shouldn't we check if `$class` is an instance of the key using `is_a` (I've no strong opinion about that).